packages/ember-runtime/lib/mixins/array.js

```diff
@@ -1724,22 +1625,11 @@ let NativeArray = Mixin.create(MutableArray, Observable, Copyable, {
   },
 
   // primitive for array support.
-  replace(idx, amt, objects) {
-    assert('The third argument to replace needs to be an array.', objects === null || objects === undefined || Array.isArray(objects));
-
-    // if we replaced exactly the same number of items, then pass only the
-    // replaced range. Otherwise, pass the full remaining array length
-    // since everything has shifted
-    let len = objects ? get(objects, 'length') : 0;
-    arrayContentWillChange(this, idx, amt, len);
+  replace(start, deleteCount, items = EMPTY_ARRAY) {
```

Comment:
The logic here slightly changed so that `null` will trigger the assertion now. I consider that a bugfix...